validator/postgresql_log_checkpoints_test.rego

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+package templates.gcp.GCPPostgreSQLCheckpointsConstraintV1
+
+import data.validator.gcp.lib as lib
+import data.validator.test_utils as test_utils
+
+import data.test.fixtures.postgresql_log_checkpoints.assets.no_settings as fixture_no_settings
+import data.test.fixtures.postgresql_log_checkpoints.assets.required as fixture_no_violation
+import data.test.fixtures.postgresql_log_checkpoints.assets.violations as fixture_violation
+
+import data.test.fixtures.postgresql_log_checkpoints.constraints as fixture_constraint
+
+template_name := "GCPPostgreSQLCheckpointsConstraintV1"
+
+#1. postgresql with correct key
+test_postgresql_log_checkpoints_required_databaseflag {
+	expected_resource_names := {"//cloudsql.googleapis.com/projects/my-test-project/instances/tf-pg-ha-62380f9c-required"}
```

Comment:
Unfortunately, this is *still* wrong.
check_test_violations_count is asserting how many violations a particular fixture (data set) should raise. According to the naming of your fixture, it should *NOT* be violating.

Yet it *is* violating, since you:
(a) Are expecting a violation (count = 1).
(b) Have the name of the supposedly correct resource in the `expected_resource_names`. `expected_resource_names` is the list of resources you expect *to raise a violation*. A correct resource should never be found here.

It seems like your logic and tests are likely *both* broken.

Please read through the full test utils and look through other tests. It's important that you get this right.